frontends/ast/simplify.cc

```diff
@@ -268,6 +263,106 @@ static AstNode *make_range(int left, int right, bool is_signed = false)
 	return range;
 }
 
+static AstNode *expand_dot(const AstNode* current_struct, const AstNode* search_node) {
+	AstNode *current_struct_elem = nullptr;
+	for (auto c : current_struct->children) {
+		auto search_str = search_node->str.find("\\") == 0 ? search_node->str.substr(1) : search_node->str;
+		if (c->str == search_str) {
+			current_struct_elem = c;
+			break;
+		}
+	}
+	log_assert(current_struct_elem);
+
+	AstNode *left = nullptr, *right = nullptr;
+	if (current_struct_elem->type == AST_STRUCT_ITEM) {
+		left = AstNode::mkconst_int(current_struct_elem->range_left, true);
+		right = AstNode::mkconst_int(current_struct_elem->range_right, true);
+	} else if (current_struct_elem->type == AST_STRUCT) {
+		// Struct can have multiple range, so to get size of 1 struct,
+		// we get left range for first children, and right range for last children
+		left = AstNode::mkconst_int(current_struct_elem->children.front()->range_left, true);
+		right = AstNode::mkconst_int(current_struct_elem->children.back()->range_right, true);
+	} else {
+		// Structs currently can only have AST_STRUCT or AST_STRUCT_ITEM
+		// so, it should never happend
+		log_assert(1 == 0);
```

Comment:
I know this shouldn't happen, but we should have a readable error message here.
Also, replace all instances of `never happend` with `never happen`.

Reply:
Done.